nbformat/validator.py

```diff
@@ -95,6 +97,84 @@ def isvalid(nbjson, ref=None, version=None, version_minor=None):
     else:
         return True
 
+def _format_as_index(indices):
+    """
+    (from jsonschema._utils.format_as_index, copied to avoid relying on private API)
+    
+    Construct a single string containing indexing operations for the indices.
+
+    For example, [1, 2, "foo"] -> [1][2]["foo"]
+    """
+
+    if not indices:
+        return ""
+    return "[%s]" % "][".join(repr(index) for index in indices)
+
+_ITEM_LIMIT = 16
+_STR_LIMIT = 64
+
+def _truncate_obj(obj):
+    """Truncate objects for use in validation tracebacks
+    
+    Cell and output lists are squashed, as are long strings, lists, and dicts.
+    """
+    if isinstance(obj, dict):
+        truncated = { k:_truncate_obj(v) for k,v in list(obj.items())[:_ITEM_LIMIT] }
+        if isinstance(truncated.get('cells'), list):
+            truncated['cells'] = ['...%i cells...' % len(obj['cells'])]
+        if isinstance(truncated.get('outputs'), list):
+            truncated['outputs'] = ['...%i outputs...' % len(obj['outputs'])]
+
+        if len(truncated) > _ITEM_LIMIT:
```

Comment:
Shouldn't this be `len(obj)`? `truncated` can't have more than _ITEM_LIMIT items, looking at the code above.